web/app/widgets/knob/knob.widget.js

```diff
@@ -112,10 +112,38 @@
                         vm.value = val;
                         OHService.sendCmd(vm.widget.item, vm.value.toString());
                     }
-                }
+                },
+                rangesEnabled: (vm.widget.rangesEnabled) ? vm.widget.rangesEnabled : false,
+                ranges: []
             }
         };
 
+
+        // if ranges are enabled update knob setings
+        if ( vm.widget.rangesEnabled) {
+          vm.widget.ranges.forEach(function(rangeItem) {
+              if (rangeItem.min != 0 || rangeItem.max != 0) {
+                  console.log("Matching:" + vm.widget.rangesTextColorMatching);
+                  if (vm.widget.rangesTextColorMatching == true) {
```

Comment:
The only difference between these conditions is `textColor` value. 
You could simplify the statement like this:
```js
var textColor = vm.widget.textColor;

if (vm.widget.rangesTextColorMatching === true) {
    textColor = rangeItem.barColor;
}

vm.knob.options.ranges.push({
   min: rangeItem.min,
   max: rangeItem.max,
   barColor: rangeItem.barColor,
   textColor: textColor
});

```